src/main/java/com/auctionapp/api/model/entities/User.java

```diff
@@ -18,119 +18,92 @@
 
-	@Column
-	private String lastName;
+    @Column
+    private String lastName;
 
-	@Column
-	private String email;
+    @Column
+    private String email;
 
-	@Column
-	private String password;
+    @Column
+    private String password;
 
-	@Column
-	private Timestamp createdAt;
+    @Column
+    private Timestamp createdAt;
 
-	@Column
-	private Timestamp updatedAt;
+    @Column
+    private Timestamp updatedAt;
 
-	@Column
-	@Enumerated(EnumType.STRING)
-	private UserRoleEnum role;
+    @Column
+    @Enumerated(EnumType.STRING)
+    private UserRole role;
 
-	public User() {
-	}
-
-	public User(final UUID id,
+    public User() {
+    }
+	
+    public User(final UUID id,
                 final String firstName,
                 final String lastName,
                 final String email,
                 final String password,
                 final Timestamp createdAt,
                 final Timestamp updatedAt,
-                final UserRoleEnum role) {
-				
-		Objects.requireNonNull(email, "The email address must not be null");
-		Objects.requireNonNull(password, "The password must not be null");
-		
+                final UserRole role) {
+
+        Objects.requireNonNull(firstName, "The first name field must not be null");
+        Objects.requireNonNull(lastName, "The last name field must not be null");
+        Objects.requireNonNull(email, "The email address must not be null");
+        Objects.requireNonNull(password, "The password must not be null");
+        Objects.requireNonNull(createdAt, "The created at field must not be null");
+        Objects.requireNonNull(updatedAt, "The updated at field must not be null");
+        Objects.requireNonNull(role, "The role field must not be null");
+
         this.id = id;
         this.firstName = firstName;
         this.lastName = lastName;
-		this.email = email;
+        this.email = email;
         this.password = password;
         this.createdAt = createdAt;
         this.updatedAt = updatedAt;
         this.role = role;
     }
 
-	public UUID getUuid() {
-		return id;
-	}
-
-	public void setUuid(UUID id) {
-		this.id = id;
-	}
-
-	public String getFirstName() {
-		return firstName;
-	}
-
-	public void setFirstName(String firstName) {
-		this.firstName = firstName;
-	}
-
-	public String getLastName() {
-		return lastName;
-	}
-
-	public void setLastName(String lastName) {
-		this.lastName = lastName;
-	}
-
-	public String getEmail() {
-		return email;
-	}
-
-	public void setEmail(String email) {
-		this.email = email;
-	}
-
-	public String getPassword() {
-		return password;
-	}
-
-	public void setPassword(String password) {
-		this.password = password;
-	}
+    public UUID getUuid() {
+        return id;
+    }
 
-	public Timestamp getCreatedAt() {
-		return createdAt;
-	}
+    public String getFirstName() {
+        return firstName;
+    }
 
-	public void setCreatedAt(Timestamp createdAt) {
-		this.createdAt = createdAt;
-	}
+    public String getLastName() {
```

Comment:
Why did you remove setters from the entity? If the user were to edit their profile information once the profile page is implemented, how would you go about that?

Reply:
Is there a reason why wouldn't I be able or maybe shouldn't add them later, once they are needed during the work on the edit user profile info ticket?

Reply:
The reason why you shouldn't is that entities should have setters, however you definitely can add it in later too from a functional standpoint. They are just very easy to add if using a tool such as IntelliJ because they can be generated automatically so there isn't really a need to wait. Either way is fine by me.